modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExchangeServiceImpl.java

```diff
@@ -170,6 +168,10 @@ private void onMessage(String nodeName, QueryBatchMessage msg) {
             } catch (Throwable e) {
                 inbox.onError(e);
 
+                if (e instanceof IgniteException) {
```

Comment:
why do we need to play such games ? Exceptions is global system machinery thus such code will bring more harm for further debugging if someone somewhere will decide to change one exception into another ? can you point me test which fail if i comment this part plz ?

Reply:
IgniteException is our own exception which we expect here and which will be processed by inbox. For any other exception we consider as unexpected error which should be processed by fail handler.
As example situation when it works - we trying insert value wider then allowed values. In this case we receive SqlException with `INTEGER out of range` (for example) and the error is expected.

To write test right now is hard unfortunately due to we can't configure fail handler yet. I will try invent how it can be tested.

Reply:
@zstan I've added test ItDmlTest#testNoFailHandlerForRuntimeSqlError